beamer/tests/contracts/test_request_manager.py

```diff
@@ -1037,6 +1037,50 @@ def test_challenge_after_l1_resolution(request_manager, token, claim_stake, cont
         request_manager.challengeClaim(claim_id)
 
 
+def test_withdraw_on_behalf(
+    request_manager, token, claim_stake, finality_period, challenge_period_extension
+):
+    first_challenger, second_challenger, requester = alloc_accounts(3)
+    (claimer,) = alloc_whitelisted_accounts(1, {request_manager})
+
+    request_id = make_request(request_manager, token, requester, requester, 1)
+    claim = request_manager.claimRequest(
+        request_id, FILL_ID, {"from": claimer, "value": claim_stake}
+    )
+    claim_id = claim.return_value
+    first_challenger_eth_balance = web3.eth.get_balance(first_challenger.address)
+    second_challenger_eth_balance = web3.eth.get_balance(second_challenger.address)
+
+    # First challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": first_challenger, "value": claim_stake + 1})
+    # Claimer outbids again
+    request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
+    # Second challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": second_challenger, "value": claim_stake + 1})
+
+    # Timetravel after claim period
+    chain.mine(timedelta=finality_period + challenge_period_extension)
+
+    request_manager.withdraw(second_challenger, claim_id, {"from": first_challenger})
```

Comment:
This is fine, but I would (also) like to test the case where a non-participant to the game calls `withdraw`.

Reply:
good point

Reply:
I tested this when withdrawing for the first challenger, so we test both cases.